src/Microsoft.Framework.Caching.Distributed/IDistributedCache.cs

```diff
@@ -1,21 +1,29 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
-using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Framework.Caching.Memory;
 
 namespace Microsoft.Framework.Caching.Distributed
 {
     public interface IDistributedCache
     {
-        void Connect();
+        byte[] Get(string key);
 
-        Stream Set(string key, object state, Action<ICacheContext> create);
+        Task<byte[]> GetAsync(string key);
 
-        bool TryGetValue(string key, out Stream value);
+        bool TryGetValue(string key, out byte[] value);
+
+        Task<bool> TryGetValueAsync(string key, out byte[] value);
+
+        void Set(string key, byte[] value, CacheEntryOptions options);
```

Comment:
Distributed cache will not use the same options type as memory cache, many of the options are different. Give it its own options type like before.